xonsh/environ.py

```diff
@@ -768,6 +768,8 @@ class Env(cabc.MutableMapping):
     def __init__(self, *args, **kwargs):
         """If no initial environment is given, os.environ is used."""
         self._d = {}
+        # sentinel value for non existing envvars
+        self._no_value = object()
```

Comment:
`None` is a valid value for an environment variable. As is `Ellipsis` and every other global singleton.

I would disagree on the "tidier" front, but that is pretty purely taste.